core/src/components/datetime/datetime.tsx

```diff
@@ -599,6 +599,14 @@ export class Datetime implements ComponentInterface {
   };
 
   private setActiveParts = (parts: DatetimeParts, removeDate = false) => {
+    /** if the datetime component is in readonly mode,
+     * allow browsing of the calendar without changing
+     * the set value
+     */
+    if (this.readonly) {
```

Comment:
The month/year picker button should be disabled. Clicking it reveals the wheel picker, and I can swipe up/down to change the month/year (which changes the `value`).

We should probably also hide the month/year picker when `readonly` is set so you don't get stuck in this state.

Reply:
i'm confused by this comment. what's being requested in this ticket is a way to navigate months in readonly mode without changing the dates. i interpreted that as being able to use both the date picker and the previous & next buttons to change the month. the other code here should be disabling the date from actually being changed when the month view changes. if you're seeing something different, could you be more specific about the reproduction? i'm not seeing the value ever get changed in readonly mode.

Reply:
In `main`, users are unable to click the month/year button to open the month/year picker when the date time is in readonly mode. This is good as opening the month/year picker would allow users to change the month and year.

https://github.com/ionic-team/ionic-framework/assets/2721089/a01241dd-6586-4a84-88f8-99f84e4dda8f

Your code regresses this behavior so users can click the month/year button to open the picker. This allows users to change the month and year which we do not want.

https://github.com/ionic-team/ionic-framework/assets/2721089/999f3ff9-81ee-46c6-940b-6b6860a32b78

I believe this issue is being caused by https://github.com/ionic-team/ionic-framework/pull/28336/files/09437b5d610388bc508f45ef7aac4255e8c32ebb#diff-83857955657186aba3b158e5561a577b94c5b9b8e74aff61fc97c78f4b77343cR201 since we are allowing pointer events.

Reply:
For reference, MUI allows the picker to be opened, but no new values can be selected from there: https://stackblitz.com/edit/react-wjqnat?file=Demo.js I don't think we should entirely disable this button, but we should prevent the value and activeParts from changing when it is used.

Reply:
I can't reproduce what you're seeing, Liam. When I change the month and/or year via the picker on my branch, the selected date does not change. 🤔

Reply:
> For reference, MUI allows the picker to be opened, but no new values can be selected from there: https://stackblitz.com/edit/react-wjqnat?file=Demo.js I don't think we should entirely disable this button, but we should prevent the value and activeParts from changing when it is used.

I don't think MUI is a great example to follow here. All of the buttons are enabled, so screen readers will consider them interactive even though they really should not be.

> I can't reproduce what you're seeing, Liam. When I change the month and/or year via the picker on my branch, the selected date does not change. 🤔

This is my code on the latest `FW-5096`:

```html
<ion-datetime readonly="true" value="2022-02-22T16:30:00"></ion-datetime>
```

1. Load the page. Observe that the button for "February 22nd, 2022" is solid blue (which indicates that it is selected).
2. Click the "February 2022" button. 
3. Click "March". 
4. Close the month/year picker. Observe that button for "March 22nd, 2022" is solid blue.

The `value` property doesn't change, but the UI still updates to indicate that March 22nd is selected which is not correct. At the very least, the wheel picker should not be interactive. However, I don't think this view should be accessible at all -- it doesn't provide any additional value for the user since the current month/year is already accessible.

Reply:
https://github.com/ionic-team/ionic-framework/assets/14926794/4f376028-ec81-495b-a36e-07c7184fb474

🤔

Reply:
> However, I don't think this view should be accessible at all -- it doesn't provide any additional value for the user since the current month/year is already accessible. 

The ticket mentions a case where multiple dates are selected across different months. The purpose of this PR is to allow you to navigate across those months so you can see all the selected dates without changing them.
(I suppose ideally, we could disable navigating across months outside of the range of the selected dates, similar to if there are min & max dates for the datetime, but I didn't think about that until now.)